tests/omprog-output-capture.sh

```diff
@@ -31,6 +35,7 @@ startup
 shutdown_when_empty
 
 # XXX
+./msleep 30000
```

Comment:
let me know if that heals the problem - if so, we may need to think about the plumbing. shutdown_when_empty should only return when the queue is really (really!) empty. I can't outrule there is a problem, though ;-)

Reply:
I don't think the problem is shutdown_when_empty. These tests do not use the feedback feature (this is on purpose, because we want to capture both stdout and stderr of the program), and therefore are susceptible to timing issues: the queue is empty, but the program has not terminated yet (the messages are buffered in the pipe), or maybe omprog has not terminated writing the captured output yet. However I had put the sleep in the wrong place I think: we must sleep before shutting down, to avoid hitting the shutdown timeout. Let's see with the next commit I just pushed...

However, if this theory is correct, I'm surprised that these timing problems appear only with FreeBSD...

The failures in Solaris are unrelated. I will skip some of the new tests from that platform because of problems with stuff used by the tests (lsof and Python). I haven't studied the causes but I think it isn't worth the effort...

Reply:
My theory wasn't correct, since adding the sleeps hasn't solved the problem. In fact, the sleeps shouldn't be necessary, since rsyslog is already waiting for the program to terminate (during 5 seconds, which is the default value of closeTimeout), and it is also waiting for the output capture thread to finish (also during 5 seconds). This should be enough, at least for the `omprog-output-capture.sh` test, which is a very simple test (but is also failing). There must be another (more serious) problem, manifesting only in FreeBSD, and likely related to the output capture thread I added in this PR.